### DISCUSSION TITLE
Problems with Social Cards for blog

### DISCUSSION THREAD

**Discussion by mpusz** [squidfunk/mkdocs-material#6583] [Q&A] [answered]
I am trying to publish a post with the new release announcement: https://mpusz.github.io/mp-units/2.2/blog/2023/12/09/mp-units-210-released/. Unfortunately, social cards do not work, and a huge image of my face (as an author) is being provided instead.

I tried to setup the social plugin explicitly:
https://github.com/mpusz/mp-units/blob/21a25013ad6624e6faff4f20f207435bf045aff7/mkdocs.yml#L64-L65

but I get the following errors in CI:

https://github.com/mpusz/mp-units/actions/runs/7397234857/job/20123979679#step:8:15

Blog-unrelated social cards seem to work fine. For example:

https://mpusz.github.io/mp-units/2.2/getting_started/introduction/

Please help 🆘

**Comment by kamilkrzyskow** [comment, marked as answer]
Hello @mpusz,
I checked both:
- https://mpusz.github.io/mp-units/2.2/blog/2023/12/09/mp-units-210-released/
- https://mpusz.github.io/mp-units/2.2/getting_started/introduction/
with the Facebook debug tool https://developers.facebook.com/tools/debug/ and the social pages work fine.
You can also see the `meta og` tags in the HTML source code.

> Unfortunately, social cards do not work

Therefore, explain where do you post the links. If it's some slack/discord chat then they might use some sort of indexed image before you enabled social cards and you should give it some time / re-shares to allow them to update the image.

> I tried to setup the social plugin explicitly:

~~All I can see there is that you added the `!ENV` with CI to the `enabled` param, which is `true` by default. Changing this won't make it more enabled than it was before.~~ My misunderstanding.

> but I get the following errors in CI:

From the error message, I see that you're using `mike` together with `sed` which replaces some version text (?) and that the warning messages point to broken links: https://mpusz.github.io/mp-units/2.2assets/images/social/blog/posts/2.0.0-released.png.
Whereas the working link is https://mpusz.github.io/mp-units/2.2/assets/images/social/blog/posts/2.0.0-released.png

Debugging `mike` is more of an upstream issue with them. Perhaps version 2.0 of `mike` has some unknown issues with the path or the social plugin doesn't deal well with it 🤔

**Reply by mpusz** [reply]
Thanks for the prompt answer. I am trying to post on LinkedIn. I do not have an account on Facebook so I can't check if it works there.

> All I can see there is that you added the !ENV with CI to the enabled param, which is true by default. Changing this won't make it more enabled than it was before.

Is the social plugin enabled implicitly, and I do not have to mention it in `mkdocs.yml`? I tried to follow https://squidfunk.github.io/mkdocs-material/plugins/social/?h=continuous+integration#config.enabled so the cache is not generated on my local machine when I develop the site.

> I see that you're using mike together with sed, which replaces some version text (?

Sed is used only to extract the version number that is later passed to mike. The extraction works correctly as I get the docs for "2.2" which is the current version of the project under development. Do you recommend submitting the issue for mike?

**Reply by mpusz** [reply]
I just tried again on LinkedIn, and it works now. It seems that the social plugin started to work only after I explicitly enabled it in `mkdocs.yml`.

**Reply by kamilkrzyskow** [reply]
For LinkedIn there is the https://www.linkedin.com/post-inspector/ and both links also work.

> I do not have to mention it in mkdocs.yml?

You do need to mention it, `social` under the `plugins` is required. I previously misunderstood the "tried to setup the social plugin explicitly" part. I thought you only added the `enabled` option, so my bad. If you added it to have different behaviour during local development and deployment, then everything is correct 👍 

> Do you recommend submitting the issue for mike?

I think so, as the https://mpusz.github.io/mp-units/latest/blog/2023/09/24/whats-new-in-mp-units-20/ address does correctly point to https://mpusz.github.io/mp-units/2.2/assets/images/social/blog/posts/2.0.0-released.png even though the link is a bit weird, as the titles differ.
Considering the actual page does have working links, where does the 404 error come from 🤔 
You can start off with running mike with the `--debug` flag as it will make the output more verbose.

**Reply by squidfunk** [reply]
@kamilkrzyskow links match the computed URL in Insiders, which contains the rewrite of the social plugin with additional capabilities:

<img width="649" alt="Bildschirm­foto 2024-01-03 um 21 12 35" src="https://github.com/squidfunk/mkdocs-material/assets/932156/8e0fb301-e210-4ca0-bb1d-51305856358a">

Nonetheless, it doesn't matter, as the image is served.
